src/metabase/query_processor/middleware/resolve_joins.clj

```diff
@@ -55,9 +55,11 @@
 ;;; |                                             :joins Transformations                                             |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
+(def ^:private default-join-alias "__join")
```

Comment:
**technically** you're allowed to specify a join **without** and alias, and previously our default alias for joins without explicit aliases was `source`. That's also our default alias for source queries (subselects). I'm surprised this hasn't caused problems in the past -- probably because 99.9% of our code specifies join aliases explicitly. I had written a TODO at some point that using the same alias for both situations was confusing/bug-prone. So while I was moving stuff around here I went ahead and changed the default join alias as well.